packages/format-library/src/collab-comment/collab-board/index.js

```diff
@@ -0,0 +1,538 @@
+	const saveComment = async () => {
+		const newComment = generateNewComment();
+		const updatedComments = getUpdatedComments( newComment, threadId );
+
+		await saveEntityRecord( 'postType', 'post', {
+			id: postId,
+			meta: {
+				collab: JSON.stringify( updatedComments ),
+			},
+		} );
+
+		setInputComment( '' );
+	};
+
+	// Function to edit the comment.
+	const editComment = async ( commentId ) => {
+		const editedComments = { ...allThreads };
+
+		if (
+			editedComments[ threadId ] &&
+			editedComments[ threadId ].comments
+		) {
+			editedComments[ threadId ].comments.map( ( comment ) => {
+				if ( comment.commentId === commentId ) {
+					comment.comment = inputComment;
+					comment.date = new Date().toISOString();
+				}
+				return comment;
+			} );
+		}
+
+		await saveEntityRecord( 'postType', 'post', {
+			id: postId,
+			meta: {
+				collab: JSON.stringify( editedComments ),
+			},
+		} );
+
+		setInputComment( '' );
+		setIsEditing( null );
+	};
+
+	// Function to mark thread as resolved
+	const markThreadAsResolved = async () => {
+		setIsResolved( true );
+
+		const updatedComments = { ...allThreads };
+
+		updatedComments[ threadId ] = {
+			...updatedComments[ threadId ],
+			isResolved: true,
+			resolvedBy: currentUser,
+			resolvedAt: new Date().toISOString(),
+		};
+
+		// Save the updated comments.
+		await saveEntityRecord( 'postType', 'post', {
+			id: postId,
+			meta: { collab: JSON.stringify( updatedComments ) },
+		} );
+
+		removeBorder();
+		setThreadId( null );
+		onClose();
+	};
+
+	// Function to delete a comment.
+	const deleteComment = async ( commentId ) => {
+		// Filter out the comment to be deleted.
+		const currentComments = allThreads[ threadId ].comments.filter(
+			( comment ) => comment.commentId !== commentId
+		);
+
+		const updatedComments = { ...allThreads };
+
+		// If there are no comments, delete the thread.
+		if ( currentComments.length === 0 ) {
+			delete updatedComments[ threadId ];
+
+			// Remove the className from the block.
+			removeBorder();
+			setThreadId( null );
+		} else {
+			updatedComments[ threadId ] = {
+				...allThreads[ threadId ],
+				comments: currentComments,
+			};
+		}
+
+		// Save the updated comments.
+		await saveEntityRecord( 'postType', 'post', {
+			id: postId,
+			meta: { collab: JSON.stringify( updatedComments ) },
+		} );
+	};
+
+	// Function to show the confirmation overlay.
+	const showConfirmationOverlay = () => setShowConfirmation( true );
+
+	// Function to hide the confirmation overlay.
+	const hideConfirmationOverlay = () => setShowConfirmation( false );
+
+	// Function to confirm and mark thread as resolved.
+	const confirmAndMarkThreadAsResolved = () => {
+		markThreadAsResolved();
+		hideConfirmationOverlay();
+	};
+
+	// Function to add a border class to the content reference.
+	const addBorder = () => {
+		if ( contentRef.current ) {
+			updateBlockAttributes( clientId, {
+				className: `block-editor-collab__${ threadId }`,
+			} );
```

Comment:
Hm, I don't think we should be modifying blocks attributes to add a border? These attributes will be saved and shown on the front-end. Not sure what this border is until I can test the functionality.

Reply:
As of now, we have implemented commenting functionality at the block level and are adding a custom class with a comment ID to block attributes.

Reply:
Right, but we really shouldn't be setting block attributes to add a class. This should be done another way, maybe with the `editor.BlockListBlock` filter or a hooks similar to [these](https://github.com/WordPress/gutenberg/blob/83f671f059531da6945e46c30f1870e1a81a2df5/packages/block-editor/src/hooks/border.js#L397C2-L397C15).

Reply:
`editor.BlockListBlock` may be more straightforward to use for now, with that you can directly filter the block props and also the class name.